### DISCUSSION TITLE
Google cast players

### DISCUSSION THREAD

**Discussion by marcelveldt** [music-assistant/support#1138] [Player Support]
Documentation https://music-assistant.io/player-support/google-cast/.

**Comment by BrilliantStanley** [comment]
Hi, is there any way to add the cast speaker manually by keying in the ip address? 
My HA server is on a different subnet, which make the speaker undiscoverable in MA.
However, the speaker can be added to HA using the ip address of it.

**Reply by marcelveldt** [reply]
No, music assistant needs to be on the same subnet as your speakers for various reason, one of them being discovery.

Just put MA on the same subnet or give it an additional network NIC.

**Comment by maxi1134** [comment]
> No, music assistant needs to be on the same subnet as your speakers for various reason, one of them being discovery.
> 
> Just put MA on the same subnet or give it an additional network NIC.


This is not true; My network is well segregated and I can detect and stream to my chromecast devices.

You do need to repeat mdns requests across VLANS and open the right ports for this.

**Reply by marcelveldt** [reply]
Repeating mdns is generally a very bad idea. It might work but it is simply not designed for it. I've also seen some very bad implementations, e.g. Unifi. You are overcomplicating your network setup. Equipment meant for simple residential network setups does not play nice with an enterprise topology. If you want to seggregate your IoT devices, better put HA/MA also in that same IoT network (you can even make HA multi homed). Opening a single TCP port for remote access to HA is way easier than rewriting mdns packets. So many examples of that working not so stable as you may wish.

**Reply by maxi1134** [reply]
I did not do extensive testing but my initial tried proved successful when using google groups. Relying on MA to sync was proven unreliable.

Since then, I've switched to Airplay2 with Wiim Pros and am testing that since 2 days ( They live on the same network as the cast devices with the same mdns repeater )

I use Unifi equipment for my house; Would you have more details on why their mdns repeater implementation is bad?

I am relatively noob in networking but strive to learn more!

**Reply by OzGav** [reply]
Marcel said it is “generally” bad. If it works all the time then sure. But if it intermittently blocks or drops messages then you are going to experience odd behaviour. We don’t have the resources to help with non standard network setups.

**Reply by teek541** [reply]
Hey guys, a few questions.

Since Google is removing the ability to have google Chromecast devices in multiple groups, I'm moving away from using google home to manage my speaker groups.

I've messed around with MA a bit, but found it to be mostly clunky and the groups confusing.  If you create a speaker group in MA will it sync the audio and does the group appear on Spotify to allow you to cast to it?

**Reply by OzGav** [reply]
No nothing else can see the groups MA creates (except HA if you are using the integration). Why do you not just use the MA interface? 
P.S. The UI is slowly being updated with improvements

**Reply by teek541** [reply]
I have spotcast and I use that to cast Spotify to my speakers.  I have a whole music dashboard that works and functions how I like it, but I want to continue to use chromecast groups.

**Reply by OzGav** [reply]
Ok I think you will need to look elsewhere. MA is a complete and self contained suite of applications.

**Comment by Tidabblju** [comment]
Where has the equalizer gone? The equalizer has simply disappeared from my two JBLs.

**Reply by OzGav** [reply]
I am assuming you mean in the advanced settings? If so, that actually didn’t do anything before as there is no API support for it so it was removed. You can make changes in the Google home app.

**Reply by Tidabblju** [reply]
I have two JBL Link where the equalizer worked very well. It's a shame if it's no longer there. In Google Home I can't change anything.

**Comment by kongo09** [comment]
Should Google cast support gapless playback? I'm using Chromecast players with Plex as source but experience a small gap between each song. I read somewhere that there is supposed to be a "flow" configuration for the players but that doesn't exist anywhere in my MA interface. What am I missing?

**Reply by OzGav** [reply]
It is in the settings for the player. Note you will lose metadata if you enable that.

**Reply by kongo09** [reply]
I don't see any player settings. If I go to Settings -> Player, this is all empty. Still I can play and see players in the main UI.

![image](https://github.com/music-assistant/hass-music-assistant/assets/640121/0b7b2c1d-ab7f-4ad5-bc50-ca14cc1f4422)

**Reply by OzGav** [reply]
Are you on the latest beta? You might have this problem https://github.com/music-assistant/hass-music-assistant/issues/1846

**Reply by kongo09** [reply]
I'm on 2.0.0b94 but I didn't try a full reinstall as advised in #1846 . Will try.

**Reply by OzGav** [reply]
Sorry for the inconvenience

**Reply by kongo09** [reply]
No worries, just trying to debug. Players appear now after reinstallation. Gapless mostly works, but I just had a funny effect where the display didn't change over to the next song but the time counter kept going up beyond the end of the song

![Screenshot_20240217-133302](https://github.com/music-assistant/hass-music-assistant/assets/640121/8cc06ce9-5ee6-4fd3-be0a-78b715e63e08)

**Reply by OzGav** [reply]
Are you using flow mode?

**Reply by kongo09** [reply]
Yes, my understanding is that enables gapless?

**Reply by OzGav** [reply]
Yeah let me ask as the way that flow mode works may result in what you saw

**Reply by kongo09** [reply]
I don't see that consistently, but I saw it when I jumped forward on the song towards near the end to test gapless.

**Reply by OzGav** [reply]
We will get you to raise an issue about this. Make sure to include all details about exactly what speaker, what player provider, and what music provider. If you can easily reproduce it then provide those steps. Do you have other players and music providers? If so, try all combinations and report the results.
